### PR TITLE
Create new feature form

### LINKED ISSUE #488: UI: Eanble create feature form
This is an ask from Trendence.  The asks is enable feature creation in Web UI. 

Currently Feature creation is already supported in backend REST API and UX mockup form is already completed. The only remaining work is integrated existing mockup form with backend api.

 Copy @chinmaytredence @xiaoyongzhu for awareness.

### PR DESCRIPTION
**Pull request by Fendoe**
Signed-off-by: Boli Guan <ifendoe@gmail.com>

## Description

Resolves #488

image

image

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to clarify your proposed changes.

### CODE REVIEW

**Comment by xiaoyongzhu** [comment]
Thanks for the PR! @Fendoe is there a staging env that folks can play with?

**Comment by Fendoe** [comment]
@xiaoyongzhu  Blair will provide it tomorrow.

**Comment by blrchen** [comment]
> Thanks for the PR! @Fendoe is there a staging env that folks can play with?

@xiaoyongzhu please checkout this staging link: https://brave-moss-045d2ce00-25.eastasia.1.azurestaticapps.net

File: ui/src/pages/NewFeature/components/FeatureForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+        <Item label="Select Transformation Type" name="selectTransformationType">
+          <Radio.Group>
+            <Space direction="vertical">
+              <Radio value={1}>Expression Transformation</Radio>
+              <Radio value={2}>Window Transformation</Radio>
+              <Radio value={3}>UDF Transformation</Radio>
+            </Space>
+          </Radio.Group>
+        </Item>
+        {selectTransformationType === 1 && (
```
Comment (blrchen) [inline ui/src/pages/NewFeature/components/FeatureForm/index.tsx:214]: Please avoid use 1,2,3 here, use an enum to make code readable.

**Comment by blrchen** [comment]
@Fendoe I did some test and find a bug for unhappy path. To repro this, just create a feature with a name that already exists in project, the api responses 400 correctly but UI displays feature created successfully message. The expected behaviour is UI should display the 400 error message returned from api to tell user detail error.

**Review by blrchen — APPROVED** [review]

**Review by Yuqing-cat — APPROVED** [review]

**Comment by xiaoyongzhu** [comment]
@Fendoe I do have a few comments and I've recorded here: https://github.com/feathr-ai/feathr/issues/955

**Pull request merged**